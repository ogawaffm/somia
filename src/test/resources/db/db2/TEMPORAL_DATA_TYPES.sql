DROP TABLE IF EXISTS TEST.TEMPORAL_DATA_TYPES;
CREATE TABLE TEST.TEMPORAL_DATA_TYPES
(
    DATE_COL         DATE,

    TIME_COL         TIME,

    -- DB2 has no second precision for TIME

    -- DB2 has no TIME WITHOUT TIME ZONE
    -- DB2 has no TIME WITH TIME ZONE

    -- default precision of 6
    TIMESTAMP_COL    TIMESTAMP,
    TIMESTAMP_0_COL  TIMESTAMP(0),
    TIMESTAMP_1_COL  TIMESTAMP(1),

    -- DB2 has no fractional seconds precision of 12
    TIMESTAMP_12_COL TIMESTAMP(12)

)