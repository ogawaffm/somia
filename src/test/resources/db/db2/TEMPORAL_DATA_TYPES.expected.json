{
  "name": "TEMPORAL_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "DATE_COL",
      "type": {
        "name": "DATE",
        "typeNumber": 91,
        "className": "java.sql.Date",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 10,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "TIME_COL",
      "type": {
        "name": "TIME",
        "typeNumber": 92,
        "className": "java.sql.Time",
        "precision": 8,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 17,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "TIMESTAMP_COL",
      "type": {
        "name": "TIMESTAMP",
        "typeNumber": 93,
        "className": "java.sql.Timestamp",
        "precision": 26,
        "scale": 6,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 46,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "TIMESTAMP_0_COL",
      "type": {
        "name": "TIMESTAMP",
        "typeNumber": 93,
        "className": "java.sql.Timestamp",
        "precision": 20,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 40,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "TIMESTAMP_1_COL",
      "type": {
        "name": "TIMESTAMP",
        "typeNumber": 93,
        "className": "java.sql.Timestamp",
        "precision": 21,
        "scale": 1,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 41,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "TIMESTAMP_12_COL",
      "type": {
        "name": "TIMESTAMP",
        "typeNumber": 93,
        "className": "java.sql.Timestamp",
        "precision": 32,
        "scale": 12,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 52,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": false,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}
